dask_expr/tests/test_collection.py

```diff
@@ -769,3 +769,21 @@ def test_align_unknown_partitions():
     df = from_pandas(pdf, npartitions=2, sort=False)
     with pytest.raises(ValueError, match="Not all divisions"):
         df.align(df)
+
+
+def test_assign_simplify(pdf):
+    df = from_pandas(pdf)
+    df2 = from_pandas(pdf)
+    df["new"] = df.x > 1
+    result = optimize(df[["x", "new"]], fuse=False)
+    expected = optimize(df2[["x"]].assign(new=df2.x > 1), fuse=False)
```

Comment:
I think I prefer using `.simplify()` instead of `optimize(..., fuse=False)` here, but either way is technically fine. (I realize we started using `optimize` in other places before `simplify` was added)

Reply:
Changed to simplify, should probably make a follow-up PR to clean this up globally.